docs/cpp.rst

```diff
@@ -0,0 +1,95 @@
+C++ Layer
+#########
+
+While GalSim is primarily a Python package, much of the implementation is done in C++
+for improved speed for the hard-core numerical calculations.
+
+If you would like to use the C++-layer functions in your own C++ project, you can do
+so with the caveat that these are officially implementation details, so we don't
+strictly enforce semantic versioning for the C++-layer API.  That is, function signatures
+may change on minor version updates (e.g. 2.3.x to 2.4.0).  We don't often make huge
+changes to the C++ API, so most of the time you will be fine upgrading, but you should
+be prepared that you may need to update your code when upgrading GalSim.  (We do
+guarantee that we won't change the C++ API for bugfix updates, e.g. 2.4.1 to 2.4.2.)
```

Comment:
API or ABI? 

In other words, can people who build their code against 2.4.1 link against 2.4.2 or do they need to recompile? If they need to recompile, then I think it is important to mention this.

Reply:
Probably mostly ABI, but I don't know really how to guarantee that.  I.e. I don't know what tests to do to ensure that we keep things ABI compatible.  There is nothing in our test suite that checks this.

Reply:
Ah, OK.  That I can do!  :)